raiden/api/rest.py

```
@@ -1219,6 +1221,37 @@ def _set_channel_reveal_timeout(
         result = self.channel_schema.dump(updated_channel_state)
         return api_response(result=result)
 
+    def _set_fee_schedule(
+        self,
+        registry_address: TokenNetworkRegistryAddress,
+        channel_state: NettingChannelState,
+        fee_schedule: FeeScheduleState,
+    ):
+        log.debug("Set fee schedule")
+        if channel.get_status(channel_state) != ChannelState.STATE_OPENED:
```

this validation should be done in `python.py`

---

From there you can raise `UnexpectedChannelState`